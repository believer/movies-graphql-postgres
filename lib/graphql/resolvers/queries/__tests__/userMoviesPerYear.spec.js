const userMoviesPerYear = require('../userMoviesPerYear')
const { verifyToken } = require('../../../../services/token')

jest.mock('../../../../services/token', () => ({
  verifyToken: jest.fn().mockImplementation(() =>
    Promise.resolve({
      id: 2,
    })
  ),
}))

describe('queries/userMoviesPerYear', () => {
  let db

  beforeEach(() => {
    db = {
      query: jest.fn().mockReturnValue('dbData'),
    }
  })

  it('should verify token', () => {
    userMoviesPerYear(null, {}, { db, token: 'test' })

    expect(verifyToken).toHaveBeenCalledWith('test')
  })

  it('should get how many movies a user has seen per year', async () => {
    await userMoviesPerYear(null, {}, { db })

    expect(db.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('should return database data', async () => {
    const response = await userMoviesPerYear(null, {}, { db })

    expect(response).toEqual('dbData')
  })

  it('should handle errors', async () => {
    db.query.mockImplementation(() => Promise.reject('nope'))

    try {
      await userMoviesPerYear(null, {}, { db })
    } catch (e) {
      expect(e.message).toMatch(/nope/)
    }
  })
})
