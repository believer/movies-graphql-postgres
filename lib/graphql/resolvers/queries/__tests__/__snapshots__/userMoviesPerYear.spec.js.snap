// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries/userMoviesPerYear should get how many movies a user has seen per year 1`] = `
"
      SELECT
        extract(year from v.view_date) as year,
        COUNT(extract(year from v.view_date))
      FROM views as v 
      WHERE v.user_id = 2
      GROUP by 1
      ORDER BY year DESC
    "
`;
